compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -733,7 +730,9 @@ object SymDenotations {
              (  !(this is Local)
              || (owner is ImplClass) // allow private local accesses to impl class members
              || isCorrectThisType(pre)
-             )
+             ) &&
```

Comment:
This long chain of conditions is getting really hard to read, factor it out into a few vals?

Reply:
I tried but it does not seem to help. The key is treating !A || B as an implication then things get clearer. I was wondering whether we should add an inline helper method `implies` on `Boolean` to do this. But it's quite a lot of effort.